udmis/src/test/java/com/google/bos/udmi/monitoring/LogTailTest.java

```diff
@@ -0,0 +1,329 @@
+    LogTail logTail = new LogTail(TEST_PROJECT_NAME);
+    return (LogTail) Mockito.spy(logTail);
+  }
+
+  @Test
+  public void testGetCloudLogStreamWhenFilter() {
+    LogTail logTailMock = getLogTailMock();
+    LoggingOptions loggingOptionsMock = Mockito.mock(LoggingOptions.class);
+    Logging loggingMock = Mockito.mock(Logging.class);
+    LogEntryServerStream streamMock = Mockito.mock(LogEntryServerStream.class);
+
+    Mockito.doReturn(loggingOptionsMock).when(logTailMock).getLoggingOptionsDefaultInstance();
+    Mockito.doReturn(loggingMock).when(loggingOptionsMock).getService();
+    Mockito.doReturn(streamMock).when(loggingMock)
+        .tailLogEntries(any(TailOption.class), any(TailOption.class));
+
+    assertEquals(streamMock, logTailMock.getCloudLogStream("meaningless filter"));
+  }
+
+  @Test
+  public void testGetCloudLogStreamWhenNoFilter() {
+    LogTail logTailMock = getLogTailMock();
+    LoggingOptions loggingOptionsMock = Mockito.mock(LoggingOptions.class);
+    Logging loggingMock = Mockito.mock(Logging.class);
+    LogEntryServerStream streamMock = Mockito.mock(LogEntryServerStream.class);
+
+    Mockito.doReturn(loggingOptionsMock).when(logTailMock).getLoggingOptionsDefaultInstance();
+    Mockito.doReturn(loggingMock).when(loggingOptionsMock).getService();
+    Mockito.doReturn(streamMock).when(loggingMock).tailLogEntries(any(TailOption.class));
+
+    assertEquals(streamMock, logTailMock.getCloudLogStream(null));
+  }
+
+  @Test
+  public void testMain() throws org.apache.commons.cli.ParseException {
+    LogTail logTailMock = getLogTailMock();
+    Mockito.doNothing().when(logTailMock).tailLogs();
+    LogTail.mainUnderTest(TEST_MAIN_ARGS, logTailMock);
+  }
+
+  @Test
+  public void testMainNoProject() {
+    String[] args = {};
+    assertThrows(RuntimeException.class, () -> LogTail.main(args));
+  }
+
+  @Test
+  public void testProcessLogEntryError() {
+    LogTail logTailMock = getLogTailMock();
+
+    LogEntry logEntryMock = Mockito.mock(LogEntry.class);
+    ProtoPayload payloadMock = Mockito.mock(ProtoPayload.class);
+    Any dataMock = Mockito.mock(Any.class);
+    ByteString byteStringMock = Mockito.mock(ByteString.class);
+    byte[] byteArray = {0};
+
+    Mockito.when(logEntryMock.getPayload()).thenReturn(payloadMock);
+    Mockito.when(payloadMock.getData()).thenReturn(dataMock);
+    Mockito.when(dataMock.getTypeUrl())
+        .thenReturn("type.googleapis.com/google.cloud.audit.AuditLog");
+    Mockito.when(dataMock.getValue()).thenReturn(byteStringMock);
+    Mockito.when(byteStringMock.toByteArray()).thenReturn(byteArray);
+
+    AuditLog auditLogMock = Mockito.mock(AuditLog.class);
+    try {
+      Mockito.doReturn(auditLogMock).when(logTailMock).getAuditLogParseFrom(any(byte[].class));
+    } catch (InvalidProtocolBufferException e) {
+      fail(e);
+    }
+
+    LogTailEntry entryTest = getLogTailEntryTest();
+    Mockito.doReturn(entryTest.timestamp).when(logEntryMock).getInstantTimestamp();
```

Comment:
(FYI you can make "doReturn" a static import and then all the leading Mockito. things go away -- doesn't make much difference, but I find it visually much cleaner)

Reply:
Sure, I made that the case for everything, to be consistent.